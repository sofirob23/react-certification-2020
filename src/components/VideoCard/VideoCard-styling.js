import styled from 'styled-components';

require('typeface-lato');

export const Title = styled.h1`
  font-family: 'Lato', sans-serif;
  font-weight: 400;
  font-size: 20px;
  padding: 5px;
`;

export const Description = styled.h1`
  font-family: 'Lato', sans-serif;
  text-align: justify;
  font-weight: 300;
  font-size: 15px;
  padding: 5px;
`;

export const Card = styled.div`
  width: 300px;
  height: 280px;
  overflow: hidden;
  display: inline-block;
  padding-bottom: 10px;
  border-radius: 10px;

  :hover {
    background-color: var(--lightgrey);
  }
`;

export const VideoPreview = styled.img`
  height: 120px;
  width: 300px;
`;
