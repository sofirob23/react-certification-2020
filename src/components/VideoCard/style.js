import styled from 'styled-components';

require('typeface-lato');

export const Title = styled.h1`
  font-family: 'Lato', sans-serif;
  font-weight: 400;
  font-size: 15px;
  padding: 5px;
  color: black;
`;

export const Description = styled.h1`
  font-family: 'Lato', sans-serif;
  text-align: justify;
  font-weight: 300;
  font-size: 10px;
  padding: 5px;
  color: black;
`;

export const Card = styled.div`
  width: 200px;
  height: 250px;
  overflow: hidden;
  display: inline-block;
  padding-bottom: 10px;
  border-radius: 10px;

  :hover {
    background-color: var(--lightgrey);
  }
`;

export const VideoPreview = styled.img`
  height: 90px;
  width: 200px;
`;
