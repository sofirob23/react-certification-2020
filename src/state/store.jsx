import React, { createContext, useReducer } from 'react';

const myStorage = window.sessionStorage;
const initialState = {
  videoList: [],
  darkMode: false,
  currentVideo: {},
  currentVideoProfile: {},
  loggedUser: myStorage.getItem('loggedUser')
    ? JSON.parse(myStorage.getItem('loggedUser'))
    : null,
  favorites: [],
};
const store = createContext(initialState);
const { Provider } = store;

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer((currentState, action) => {
    switch (action.type) {
      case 'search':
        return {
          videoList: action.payload,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'darkMode':
        return {
          videoList: currentState.videoList,
          darkMode: action.payload,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'play':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: action.video,
          currentVideoProfile: action.profile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'login':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: action.payload,
          favorites: currentState.favorites,
        };
      case 'logout':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: null,
          favorites: currentState.favorites,
        };
      case 'addFavorite':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites.append(action.payload),
        };
      case 'removeFavorite':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites.filter(
            (video) => video.id.videoId !== action.payload.id.videoId
          ),
        };
      default:
        throw new Error();
    }
  }, initialState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

export { store, StateProvider };
