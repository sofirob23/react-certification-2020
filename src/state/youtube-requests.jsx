import axios from 'axios';

const search = {
  part: 'snippet',
  key: process.env.REACT_APP_API_KEY,
};

const youtubeSearch = (requestParams) => {
  const youtube = axios.create({
    baseURL: 'https://www.googleapis.com/youtube/v3/',
    params: Object.assign(search, requestParams),
  });
  return youtube.get('/search');
};

export default youtubeSearch;
