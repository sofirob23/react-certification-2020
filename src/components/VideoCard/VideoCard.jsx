import React from 'react';

import { Card, Description, Title, VideoPreview } from './VideoCard-styling';

const VideoCard = (video) => {
  return (
    <Card>
      <VideoPreview src={video.video.snippet.thumbnails.medium.url} />
      <Title>{video.video.snippet.title}</Title>
      <Description>{video.video.snippet.description}</Description>
    </Card>
  );
};

export default VideoCard;
