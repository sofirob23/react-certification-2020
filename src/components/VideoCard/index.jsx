import React from 'react';

import { Card, Description, Title, VideoPreview } from './style';

const VideoCard = (props) => {
  return (
    <Card>
      <VideoPreview src={props.video.snippet.thumbnails.medium.url} />
      <Title>{props.video.snippet.title}</Title>
      <Description>{props.video.snippet.description}</Description>
    </Card>
  );
};

export default VideoCard;
