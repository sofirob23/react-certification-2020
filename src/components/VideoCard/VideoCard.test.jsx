import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Router, StaticRouter } from 'react-router';
import { createMemoryHistory } from 'history';
import videoMock from '../../utils/mocks/video-mock.json';
import { store } from '../../state/store';
import VideoCard from './index';

describe('Video Card Tests', () => {
  const { Provider } = store;
  const dispatch = jest.fn();
  const state = {
    videoList: [],
    darkMode: false,
    currentVideo: videoMock,
    currentVideoProfile: {},
  };

  test('Video Card has image title and description', () => {
    render(
      <Provider value={{ dispatch, state }}>
        <StaticRouter>
          <VideoCard video={videoMock} />
        </StaticRouter>
      </Provider>
    );
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByText(videoMock.snippet.title)).toBeInTheDocument();
    expect(screen.getByText(videoMock.snippet.description)).toBeInTheDocument();
  });

  test('Video Card redirects on Click', () => {
    const history = createMemoryHistory();
    render(
      <Provider value={{ dispatch, state }}>
        <Router history={history}>
          <VideoCard video={videoMock} />
        </Router>
      </Provider>
    );
    fireEvent.click(screen.getByRole('img'));
    const newPath = `/video/${videoMock.id.videoId}`;
    expect(history.location.pathname).toBe(newPath);
  });

  test('Video Card from favorites contains icon', () => {
    render(
      <Provider value={{ dispatch, state }}>
        <StaticRouter>
          <VideoCard video={videoMock} favorites />
        </StaticRouter>
      </Provider>
    );
    expect(screen.getByTestId('favorite-icon')).toBeInTheDocument();
  });

  test('Video Card favorite redirects to favorites on Click', () => {
    const history = createMemoryHistory();
    render(
      <Provider value={{ dispatch, state }}>
        <Router history={history}>
          <VideoCard video={videoMock} favorites />
        </Router>
      </Provider>
    );
    fireEvent.click(screen.getByRole('img'));
    const newPath = `/favorites/${videoMock.id.videoId}`;
    expect(history.location.pathname).toBe(newPath);
  });
});
