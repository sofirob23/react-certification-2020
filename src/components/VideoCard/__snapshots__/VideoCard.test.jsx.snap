// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Card Tests Video Card renders correctly 1`] = `
<div
  className="sc-dlfnbm gOyLuw"
>
  <img
    className="sc-hKgILt cjtqHK"
    src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s240-c-k-c0xffffffff-no-rj-mo"
  />
  <h1
    className="sc-bdfBwQ clmpDO"
  >
    Wizeline
  </h1>
  <h1
    className="sc-gsTCUz bPkPNX"
  >
    Wizeline transforms how teams build technology. Its customers accelerate the delivery of innovative products with proven solutions, which combine Wizeline's ...
  </h1>
</div>
`;
